daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/TDMLRunner.scala

```diff
@@ -429,6 +429,8 @@ class DFDLTestSuite private[tdml] (
 abstract class TestCase(testCaseXML: NodeSeq, val parent: DFDLTestSuite)
   extends Logging {
 
```

Comment:
Add comment that TestCase execution is strongly sequential and single-thread because whenever we create a new processor with new settings we just assign this var. 

I did things this way rather than restructure the code more functionally, which would have been a lot of churn. So this code still reflects imperative style code of the API it was using before.